controllers/csi/metadata/sqlite.go

```diff
@@ -0,0 +1,306 @@
+package metadata
```

Comment:
I think it would be a better solution to use a map or a custom resource to achieve this instead of bringing the complexity of this in there ...

Reply:
Using maps (tenants/volume map) could work well up until `GetUsedVersions(tenantUUID string)`, you would have to iterate over all of the volume-map to find the versions associated with a tenant, a 3rd map just for the purpose of storing what versions does a tenant have would be yet again a bit annoying because you can't really store a key: list AFAIK (you can store the list as a string) in a configmap.

A smaller issue is that this would create 2(or 3) configmap/node, which feels a bit much 😄 
And creating a custom resource to solve this issue feels a bit more complex to me.

however, I'm open for suggestions 😃